backend/fms_core/migrations/0069_v5_0_0.py

```diff
@@ -0,0 +1,76 @@
+import reversion
+import re
+
+from django.db import migrations, models
+import django.db.models.deletion
+from django.contrib.auth.models import User
+
+ADMIN_USERNAME = 'biobankadmin'
+
+def set_dataset_project_fk_using_project_name(apps, schema_editor):
+    Project = apps.get_model("fms_core", "Project")
+    Dataset = apps.get_model("fms_core", "Dataset")
+
+    with reversion.create_revision(manage_manually=True):
+        admin_user = User.objects.get(username=ADMIN_USERNAME)
+
+        reversion.set_comment("Insert fk of FMS project for each existing Dataset before removing project related fields.")
+        reversion.set_user(admin_user)
+
+        for dataset in Dataset.objects.all():
+            try:
+                project = Project.objects.get(name=dataset.project_name)
+            except:
+                project = Project.objects.get(name__startswith=dataset.project_name)
```

Comment:
Filter with `name__startswith` seems kinda risky.

Reply:
A bit in the sense it might not work but if we don't get exactly one match it is going to fail. The issue here is that someone had the great idea to rename a project, the good news is the renaming was done by appending stuff ... This is mainly a single project causing issue to the migration. I will have to run the migration on a prod backup the day before the release to make sure all still runs ...